_episodes/14-writing-functions.md

```diff
@@ -264,82 +262,77 @@ result of call is: None
 
 > ## Encapsulating Data Analysis
 >
-> Assume that the following code has been executed:
+> Write a method to determine the average GDP for a country in a specific decade from the gapminder_gdp data.
+> 1. Fill in the blanks to determine the average GDP for Japan in the 1980's
 >
+> (Note '//' is the symbol for division but returns a whole number)
 > ~~~ 
 > import pandas
 >
 > df = pandas.read_csv('gapminder_gdp_asia.csv', index_col=0)
 > japan = df.ix['Japan']
-> ~~~
-> {: .python}
->
-> 1. Complete the statements below to obtain the average GDP for Japan
->    across the years reported for the 1980s.
->
-> ~~~ 
-> year = 1983
-> gdp_decade = 'gdpPercap_' + str(year // ____)
+> searchyear = 1983
+> gdp_decade = 'gdpPercap_' + str(searchyear // ____)
 > avg = (japan.ix[gdp_decade + ___] + japan.ix[gdp_decade + ___]) / 2
 > ~~~
 > {: .python}
 >
-> 2. Abstract the code above into a single function.
+> 2. Abstract the code above into a single function to allow any country and any decade.
 >
 > ~~~
-> def avg_gdp_in_decade(country, continent, year):
->     df = pd.read_csv('gapminder_gdp_'+___+'.csv',delimiter=',',index_col=0)
->     ____
->     ____
->     ____
->     return avg
+> def avg_gdp_in_decade(country, continent, searchyear):
+>     df = pd.read_csv('gapminder_gdp_' + ___ + '.csv',delimiter=',',index_col=0)
+>     country_gdp = df.ix[___]
+>     startyear = (searchyear // __) * __
+>     sum_gdp = 0
+>     count_gdp = 0
+>     for year in range(startyear, startyear + __):
+>         decade = ________
+>         if decade in country_gdp:
+>             sum_gdp = ________
+>             count_gdp = ________
+>     avg_gdp = ________ / ________
+>     print("Avg for ", country, " in the ", startyear, "'s is ", avg_gdp)
+>     return avg_gdp
 > ~~~
 > {: .python}
 >
-> 3. How would you generalize this function
->    if you did not know beforehand which specific years occurred as columns in the data?
->    For instance, what if we also had data from years ending in 1 and 9 for each decade?
->    (Hint: use the columns to filter out the ones that correspond to the decade,
->    instead of enumerating them in the code.) 
->
 > > ## Solution
 > >
-> > 1. 
+> > 1. This requires knowledge of the column headings so is of limited re-use
```

Comment:
I think there are a few problems that are scattered through the lessons that use column headings.  I thinnk we should retain this variation. (Not all of the exercises are required of course.)